### PR TITLE
Index ANS v2

### PR DESCRIPTION
**Pull request by rtso**
## Description
Adds support for ANS v2 indexing. The indexed data is output to the same tables, so no changes will need to be made by end users. 

config.yaml
```
# Testnet
starting_version: 684281798 # txn version when ANS v2 was published
ans_processor_config:
      ans_v1_primary_names_table_handle: "0x0130568e434c6d73a31b07cbd9909f29012e05c8954101960930bd9b59355663"
      ans_v1_name_records_table_handle: "0xe0a2be534a7b8b31fea6556ac8a84bf9fb5f06f4b6453cd98f0edc184deef49e"
      ans_v2_contract_address: "0x5f8fd2347449685cf41d4db97926ec3a096eaf381332be4f1318ad4d16a8497c"

# Mainnet
starting_version: 2070800 # ANS contract was first deployed on mainnet at transaction 2070800
ans_processor_config:
      ans_v1_primary_names_table_handle: "0x1d5f57aa505a2fa463b7a46341913b65757e3177c46a5e483a29d953627bee62"
      ans_v1_name_records_table_handle: "0x21a0fd41330f3a0a38173c7c0e4ac59cd51505f0594f64d3d637c12425c3c155"
      ans_v2_contract_address: "0x867ed1f6bf916171b1de3ee92849b8978b7d1b9e0a8cc982a3d19d535dfd9c0c"

```

### ANS activities table
Screenshot 2023-09-22 at 3 04 44 PM
Screenshot 2023-09-22 at 2 49 55 PM

### ANS current table
Screenshot 2023-09-22 at 3 06 02 PM
Screenshot 2023-09-22 at 3 05 54 PM

### current_aptos_names view
Screenshot 2023-09-22 at 3 19 22 PM

### CODE REVIEW

File: rust/processor/src/worker.rs [hunk 1]
```diff
@@ -705,6 +702,7 @@ pub async fn create_fetcher_loop(
                 tokio::time::sleep(std::time::Duration::from_millis(100)).await;
             }
             info!("[Parser] The stream is ended.");
+            break;
```
Comment (rtso) [inline rust/processor/src/worker.rs:705]: Need this or else code will infinite loop if `ending_version` is set

File: rust/processor/src/processors/ans_processor.rs [hunk 2]
```diff
@@ -296,95 +335,193 @@ fn parse_ans(
...
+                .unwrap()
+                {
+                    migration_burn_events.insert(migration_burn_event.burned_v1_token_name);
+                }
+            }
+
+            // Parse V1 ANS write set changes
+            for (wsc_index, wsc) in transaction_info.changes.iter().enumerate() {
+                match wsc.change.as_ref().unwrap() {
+                    // TODO: Don't index v1 changes if token burn detected
```
Comment (bowenyang007) [inline rust/processor/src/processors/ans_processor.rs:394]: why not?

File: rust/processor/src/processors/ans_processor.rs [hunk 3]
```diff
@@ -296,95 +335,193 @@ fn parse_ans(
...
+                                    error = ?e,
+                                    "Error parsing ANS v1 name record from write table item"
+                                );
+                                panic!();
+                            })
+                        {
+                            // Don't index this v1 wsc if it's been migrated to v2
+                            if migration_burn_events.contains(&current_ans_lookup.token_name) {
+                                continue;
+                            }
```
Comment (bowenyang007) [inline rust/processor/src/processors/ans_processor.rs:414]: why not?

Comment (rtso) [reply rust/processor/src/processors/ans_processor.rs:414]: When a v1 token migrates to v2, the data in the v1 table also changes (for example, the v1 token's target address gets unset [Example txn](https://explorer.aptoslabs.com/txn/663354522/changes?network=local)). So we don't want to index the v1 data if we detect it's been migrated, we only want to index the v2 data.

File: rust/processor/src/processors/ans_processor.rs (hunk 2 above)
Comment (rtso) [reply rust/processor/src/processors/ans_processor.rs:394]: I explained below

File: rust/processor/migrations/2023-09-08-161603_add_ans_v2/up.sql [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  token_name VARCHAR(140),
+  registered_address VARCHAR(66),
+  expiration_timestamp TIMESTAMP NOT NULL,
+  last_transaction_version BIGINT NOT NULL,
+  is_deleted BOOLEAN NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (domain, subdomain, token_standard)
+);
+CREATE INDEX IF NOT EXISTS ans_v2_ts_index on current_ans_lookup_v2 (token_standard);
```
Comment (bowenyang007) [inline rust/processor/migrations/2023-09-08-161603_add_ans_v2/up.sql:16]: why? Let's not add redundant indices, especially if this one has only a cardinality of 2.

**Review by bowenyang007 — APPROVED** [review]
Took a cursory look but overall looks good!

**Comment by kent-white** [comment]
Could we get another field here for if the token is v1 or v2?

**Review by kent-white — APPROVED** [review]

**Pull request merged**